Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orientation.h

```diff
@@ -334,7 +336,312 @@ void reverse_face_orientations(const FaceRange& face_range, PolygonMesh& pmesh)
+    Polygon_mesh_processing::reverse_face_orientations(cc_faces, tm);
+    new_is_parent_outward_oriented = !new_is_parent_outward_oriented;
+  }
+  cc_handled.set(xtrm_cc_id);
+
+  std::size_t nb_cc = cc_handled.size();
+
+// get all cc that are inside xtrm_cc_id
+
+  typename Side_of_tm::AABB_tree aabb_tree(cc_faces.begin(), cc_faces.end(),
+                                           tm, vpm);
+  Side_of_tm side_of_cc(aabb_tree);
+
+  std::vector<std::size_t> cc_inside;
+  for(std::size_t id=0; id<nb_cc; ++id)
+  {
+    if (cc_handled.test(id)) continue;
+    if (side_of_cc(get(vpm,xtrm_vertices[id]))==ON_BOUNDED_SIDE)
+      cc_inside.push_back(id);
+  }
+
+// check whether we need another recursion for cc inside xtrm_cc_id
+  if (!cc_inside.empty())
+  {
+    std::size_t new_xtrm_cc_id = cc_inside.front();
+    boost::dynamic_bitset<> new_cc_handled(nb_cc,0);
+    new_cc_handled.set();
+    new_cc_handled.reset(new_xtrm_cc_id);
+    cc_handled.set(new_xtrm_cc_id);
+
+    std::size_t nb_candidates = cc_inside.size();
+    for (std::size_t i=1;i<nb_candidates;++i)
+    {
+      std::size_t candidate = cc_inside[i];
+      if(get(vpm,xtrm_vertices[candidate]).z() >
+         get(vpm,xtrm_vertices[new_xtrm_cc_id]).z()) new_xtrm_cc_id=candidate;
+      new_cc_handled.reset(candidate);
+      cc_handled.set(candidate);
+    }
+
+    internal::recursive_orient_volume_ccs<Kernel>(
+           tm, vpm, fid_map, xtrm_vertices, new_cc_handled, face_cc,
+           new_xtrm_cc_id, new_is_parent_outward_oriented);
+  }
+
+// now explore remaining cc included in the same cc as xtrm_cc_id
+  boost::dynamic_bitset<> cc_not_handled = ~cc_handled;
+  std::size_t new_xtrm_cc_id = cc_not_handled.find_first();
+  if (new_xtrm_cc_id == cc_not_handled.npos) return ;
+
+  for (std::size_t candidate = cc_not_handled.find_next(new_xtrm_cc_id);
+                   candidate < cc_not_handled.npos;
+                   candidate = cc_not_handled.find_next(candidate))
+  {
+     if(get(vpm,xtrm_vertices[candidate]).z() > get(vpm,xtrm_vertices[new_xtrm_cc_id]).z())
+        new_xtrm_cc_id = candidate;
+  }
+
+  internal::recursive_orient_volume_ccs<Kernel>(
+            tm, vpm, fid_map, xtrm_vertices, cc_handled, face_cc,
+            new_xtrm_cc_id, is_parent_outward_oriented);
+}
+
+}//end internal
+
+/**
+* \ingroup PMP_orientation_grp
+* makes each connected component of a closed triangulated surface mesh
+* inward or outward oriented.
+*
+* @tparam TriangleMesh a model of `FaceListGraph` and `MutableFaceGraph` .
+*                      If `TriangleMesh` has an internal property map for `CGAL::face_index_t`,
+*                      as a named parameter, then it must be initialized.
+* @tparam NamedParameters a sequence of \ref namedparameters
+*
+* @param tm a closed triangulated surface mesh
+* @param orient_outward `true` indicates that each connected component will be outward oriented,
+*                        (inward oriented if `false`). Default value is true.
```

Comment:
if `orient_outward` has a valid default value, why not integrate it to the named parameters? (in both functions)